From e798fed5789deb5d4571993d7da7494647525adc Mon Sep 17 00:00:00 2001
From: Amit Prakash Ambasta <amit.prakash.ambasta@gmail.com>
Date: Mon, 6 Jun 2022 16:53:05 +0530
Subject: [PATCH] widget/gtk/mozgtk.c:

  Added a dummy void mozgtk_ensure_linked() {} function outside the
  MOZ_X11 ifdef blocks for linkers to ensure using mozgtk when linking
  with --as-needed

widget/gtk/nsWindow.cpp:

  Added a call to mozgtk_ensure_linked() to ensure linkage when linking
  with --as-needed

Signed-off-by: Amit Prakash Ambasta <amit.prakash.ambasta@gmail.com>
---
 widget/gtk/mozgtk/mozgtk.c | 2 ++
 widget/gtk/nsWindow.cpp    | 1 +
 2 files changed, 3 insertions(+)

diff --git a/widget/gtk/mozgtk/mozgtk.c b/widget/gtk/mozgtk/mozgtk.c
index d7dbbc6fd9..ae8662e3c4 100644
--- a/widget/gtk/mozgtk/mozgtk.c
+++ b/widget/gtk/mozgtk/mozgtk.c
@@ -6,6 +6,8 @@
 
 #include "mozilla/Types.h"
 
+MOZ_EXPORT void mozgtk_ensure_linked() {}
+
 #ifdef MOZ_X11
 #  include <X11/Xlib.h>
 // Bug 1271100
diff --git a/widget/gtk/nsWindow.cpp b/widget/gtk/nsWindow.cpp
index c21ed38565..54d8395477 100644
--- a/widget/gtk/nsWindow.cpp
+++ b/widget/gtk/nsWindow.cpp
@@ -5314,6 +5314,7 @@ void nsWindow::ConfigureGdkWindow() {
 #ifdef MOZ_WAYLAND
   if (GdkIsWaylandDisplay()) {
     mSurfaceProvider.Initialize(this);
+    mozgtk_ensure_linked();
   }
 #endif
 
-- 
2.35.1

